### ISSUE TITLE
tiling to analyze large images

### ISSUE THREAD

**Issue by dalessioluca** [spacetx/spacetx-research#9]
def tiling(img, crop_w, crop_h, stride_w, strid_h):
n_obj = 0 
integer_segmentation_mask = torch.zeros_like(img)
vae.eval()
for crop in crops:
     out = vae.forward(crop)
     segmentation = out.inference.integer_segmentation_mask

For each crop it gets the integer_segmentation_mask which:
1. is censured outside the region which do not suffers of the boundary effect (how do I do this?)
2. shifted by the number of instances already found, i.e.  mask = torch.where(mask > 0, mask+shift, 0) 
3. pasted in the right place

**Comment by dalessioluca** [comment]
Read: https://towardsdatascience.com/review-u-net-biomedical-image-segmentation-d02bf06ca760 for tiling strategy

**Comment by dalessioluca** [comment]
In function segment_with_tiling I have the line:
                    integer_mask = integer_mask * filter_keep  # this need to be improved
This need to be improved b/c it chops objects that extend across tiling boundaries

**Comment by dalessioluca** [comment]
I think your idea of building a pixel-pixel graph makes a lot of sense
It is precisely our "co-objectness" graph!
But this time, we're getting the probs from actually being detected in the same object!
By the way, we can build the full graph (and not just NN graph) by populating a COO scipy sparse matrix
Iterate through every detected instance, and +1 to all pairs of pixels that belong to that instance
Since the objects are small and localized, the graph remains sparse
You can build a COO sparse matrix by providing a list of row indices, col indices, and value; repeated row and col indices are allowed -- the values will be added together
